reactors/1D_packbed.ipynb

```diff
@@ -0,0 +1,518 @@
+   "metadata": {},
+   "source": [
+    "### Define reactor geometry and operating conditions"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Modeling packed-bed, catalytic-membrane reactor...\n",
+      "H2 permeable membrane is present.\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Reactor geometry\n",
+    "L = 5e-2  # length of the reactor (m)\n",
+    "R = 5e-3  # radius of the reactor channel (m)\n",
+    "phi = 0.5  # porosity of the bed (-)\n",
+    "tau = 2.0  # toruosity of the bed (-)\n",
+    "dp = 3.37e-4  # particle diameter (m)\n",
+    "As = 3.5e6  # specific surface area (1/m)\n",
+    "\n",
+    "# Energy (adiabatic or isothermal)\n",
+    "solve_energy = True  # True: Adiabatic, False: isothermal\n",
+    "\n",
+    "# Membrane\n",
+    "membrane_present = (\n",
+    "    True  # flag to include membrane (False = no membrane, True = membrane)\n",
+    ")\n",
+    "membrane_perm = 1e-15  # membrane permeability (kmol*m3/s/Pa)\n",
+    "thickness = 3e-6  # membrane thickness (m)\n",
+    "membrane_sp_name = \"H2\"  # membrane-permeable species name\n",
+    "p_sweep = 1e5  # partial pressure of permeable species in the sweep channel (Pa)\n",
+    "permeance = membrane_perm / thickness  # permeance of the membrane (kmol*m2/s/Pa)\n",
+    "\n",
+    "if membrane_present == True:\n",
+    "    print(\"Modeling packed-bed, catalytic-membrane reactor...\")\n",
+    "    print(membrane_sp_name, \"permeable membrane is present.\")\n",
+    "\n",
+    "# Get required properties based on the geometry and mechanism\n",
+    "W_g = gas.molecular_weights  # vector of molecular weight of gas species\n",
+    "vol_ratio = phi / (1 - phi)\n",
+    "eff_factor = phi / tau  # effective factor for permeability calculation\n",
+    "B_g = (\n",
+    "    pow(vol_ratio, 2) * pow(dp, 2) * eff_factor / 72\n",
+    ")  # permeability based on Kozeny-Carman equation\n",
+    "area2vol = 2 / R  # area to volume ratio assuming a cylindrical reactor\n",
+    "D_h = 2 * R  # hydraulic diameter\n",
+    "membrane_sp_ind = gas.species_index(\n",
+    "    membrane_sp_name\n",
+    ")  # index of the membrane-permeable species\n",
+    "\n",
+    "# Inlet operating conditions\n",
+    "T_in = 673  # inlet temperature [K]\n",
+    "p_in = 5e5  # inlet pressure [Pa]\n",
+    "v_in = 0.001  # inlet velocity [m/s]\n",
+    "T_wall = 723  # wall temperature [K]\n",
+    "h_coeff = 1e2  # heat transfer coefficient\n",
+    "\n",
+    "# Set gas and surface states\n",
+    "gas.TPX = T_in, p_in, \"NH3:0.99, AR:0.01\"  # inlet composition\n",
+    "surf.TP = T_in, p_in\n",
+    "Yk_0 = gas.Y\n",
+    "rhou0 = gas.density * v_in\n",
+    "\n",
+    "# Initial surface coverages\n",
+    "surf.advance_coverages(\n",
+    "    1e10\n",
+    ")  # advancing coverages over a long period of time to get the steady state.\n",
```

Comment:
Rather than having end-of-line comments that cause very strange breaks like this, you can put the comment above the code.

Reply:
Corrected.